src/GovUk.Education.ExploreEducationStatistics.Content.Services.Tests/DataSetServiceTests.cs

```diff
@@ -0,0 +1,205 @@
+
+                var result = await dataSetService.GetDataSet(
+                    publication.Releases[0].Id, Guid.NewGuid());
+
+                result.AssertNotFound();
+            }
+        }
+
+        [Fact]
+        public async Task ReleaseNotPublished()
+        {
+            var publication = _fixture.DefaultPublication()
+                .WithReleaseParents(
+                    _fixture.DefaultReleaseParent(publishedVersions: 0, draftVersion: true, year: 2000)
+                        .Generate(1))
+                .WithTopic(_fixture.DefaultTopic()
+                    .WithTheme(_fixture.DefaultTheme()))
+                .Generate();
+
+            var releaseFile = _fixture.DefaultReleaseFile()
+                .WithRelease(publication.Releases[0])
+                .WithFile(_fixture.DefaultFile())
+                .Generate();
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.ReleaseFiles.AddRangeAsync(releaseFile);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var dataSetService = BuildService(contentDbContext);
+
+                var result = await dataSetService.GetDataSet(
+                    publication.Releases[0].Id, Guid.NewGuid());
+
+                result.AssertNotFound();
+            }
+        }
+
+        [Fact]
+        public async Task AmendmentNotPublished()
+        {
+            var publication = _fixture.DefaultPublication()
+                .WithReleaseParents(
+                    _fixture.DefaultReleaseParent(publishedVersions: 1, draftVersion: true, year: 2000)
+                        .Generate(2))
+                .WithTopic(_fixture.DefaultTopic()
+                    .WithTheme(_fixture.DefaultTheme()))
+                .Generate();
+
+            var releaseFile = _fixture.DefaultReleaseFile()
+                .WithRelease(publication.Releases[1])
+                .WithFile(_fixture.DefaultFile())
+                .Generate();
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.ReleaseFiles.AddRangeAsync(releaseFile);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var dataSetService = BuildService(contentDbContext);
+
+                var result = await dataSetService.GetDataSet(
+                    publication.Releases[0].Id, Guid.NewGuid());
+
+                result.AssertNotFound();
+            }
+        }
+    }
```

Comment:
I think I would leave the previous test to test an unpublished version which it's already doing, and change this into one which tests a release which isn't the latest published version.

It would still test `publication.Releases[0].Id` but you would give it `publishedVersions: 2` and it could also have a draft version.

You would set up the release file to be on all 3 versions.

If you don't do this, you haven't got a test which checks you can't get a data set that exists on a previously published  version.

Reply:
Reviewed latest changes and I still think it's a bit confusing as its trying to be both a "FileIsNotAssociatedWithRelease" and a "ReleaseIsNotLatestPublishedVersion" test at the same time.

You could argue you don't need the first of these cases as you've already got a "NoFile" test.

For the latter case I still think it would be better if the test set up the same file for all versions.

All other changes seem fine btw.

A blank line has crept in on L102.

I didn't mention this earlier but there's a style guide pointer now about test method names now following `Scenario_Result` pattern with  the simple happy-path scenario being called `Success` for succinctness. I'm not sure if this means some of your tests here should be suffixed with `_ReturnsNotFound` or not, e.g. `NoFile` - `NoFile_ReturnsNotFound`.